Composer/packages/server/src/models/bot/botProject.ts

```diff
@@ -728,6 +729,46 @@ export class BotProject implements IBotProject {
       fileList.set(file.name, file);
     });
 
+    const migrationFiles = await this._createQnAFilesForOldBot(fileList);
+
+    return new Map<string, FileInfo>([...fileList, ...migrationFiles]);
+  };
+
+  // migration: create qna files for old bots
+  private _createQnAFilesForOldBot = async (files: Map<string, FileInfo>) => {
```

Comment:
I like that this work has been moved to the server to create the qna files for the bot. The code that was replaced in the DesignPage created a new qna file when a new dialog file is created on the client. It looks like this will only handle existing dialogs of the bot.

Reply:
The qna files for the new dialogs will are handled in dispatcher/dialogs.

Reply:
![216db8343122f5dbac554463c1fde8c](https://user-images.githubusercontent.com/24380525/93546867-600b8600-f996-11ea-9ca0-b45045ece921.png)
In dispatchers/dialog.ts line 78. It already handled it.